### PR TITLE
Add 'allowed repositories' description to docs

### PR DESCRIPTION
**Pull request by gilesgas**
This PR was created to address the https://github.com/buildkite/docs/issues/2646 issue of the Buildkite docs, stemming from this PR https://github.com/buildkite/agent/pull/2361 of the agent codebase.

I recently discovered that the docs location for some table entries in the Buildkite docs is housed in YAML files.

Therefore, based on the information I gathered from [this section of the agent codebase](https://github.com/buildkite/agent/pull/2361/files#diff-978b503ab54eee245e9d9de4050c039219fe264fc02958e53a1d36e8ea097a20R544), I created  this PR to update the docs.

### CODE REVIEW

File: data/content/agent_attributes.yaml [hunk 1]
```diff
@@ -4,19 +4,25 @@ attributes:
...
+  default_value: "none"
+  required: false
+  desc: |
+      A comma-separated list of regular expressions representing repositories the agent is allowed to clone (for example, "^git@github.com:buildkite/.\*" or "^https://github.com/buildkite/.*")
+- name: bootstrap-script
+  env_var: BUILDKITE_BOOTSTRAP_SCRIPT_PATH
+  default_value: "buildkite-agent bootstrap"
+  required: false
+  desc: |
+      Command to invoke the bootstrap process.
```
Comment (gilesgas) [inline data/content/agent_attributes.yaml:25]: I decided to move this entry on `bootstrap-script` here, as it better matches how these entries appear in the table in the docs - that is, mandatory attributes at the start, followed by optional (in alphabetical order).

File: data/content/agent_attributes.yaml [hunk 2]
```diff
@@ -4,19 +4,25 @@ attributes:
...
+      The agent registration token from your organization's _Agents_ page, used only to register new agents. To get your token, log in to Buildkite, navigate to _Agents_, then select _Reveal Agent Token_.
 - name: build-path
   env_var: BUILDKITE_BUILD_PATH
   default_value: "(depends on platform)"
   required: true
   desc: |
       Path to where the builds will run from.
+- name: allowed-repositories
+  env_var: BUILDKITE_ALLOWED_REPOSITORIES
+  default_value: "none"
```
Comment (gilesgas) [inline data/content/agent_attributes.yaml:16]: Question, is the `default_value` actually equal to `"none"`?

Comment (moskyb) [reply data/content/agent_attributes.yaml:16]: nah, the default value will be `""`

Comment (gilesgas) [reply data/content/agent_attributes.yaml:16]: Thanks for that, sounds sensible!

**Pull request merged**